src/coreclr/jit/lclvars.cpp

```diff
@@ -4348,6 +4347,7 @@ void Compiler::lvaMarkLocalVars()
             lvaPSPSym            = lvaGrabTempWithImplicitUse(false DEBUGARG("PSPSym"));
             LclVarDsc* lclPSPSym = &lvaTable[lvaPSPSym];
             lclPSPSym->lvType    = TYP_I_IMPL;
+            lvaSetVarDoNotEnregister(lvaPSPSym DEBUGARG(DoNotEnregisterReason::VMNeedsStackAddr));
```

Comment:
now this is the main reason for `VMNeedsStackAddr` but if I understand correctly we actually report this to VM so we can't do anything about it.